### DISCUSSION TITLE
CKEditor Title Attributes for Anchor Tags

### DISCUSSION THREAD

**Discussion by jmhall-vabeach** [craftcms/ckeditor#167] [Ideas] [answered]
Federal Guidelines require municipalities to ensure that all site content is accessible. With that said we are required by law to add title attributes to anchor tags. I have yet to find this in CKEditor. The only option I see is a downloadable option. Will you please consider adding the same functionality that Redactor has for adding title support? It would be great if you made it required.

**Comment by gcamacho079** [comment]
@jmhall-vabeach, when you refer to federal guidelines to ensure site content is accessible, do you mean conformance with WCAG 2.2 AA?

I’m unaware of any hard requirement to add `title` attributes to anchor tags, as adding `title` to links (the [WCAG 2.2 Technique H33: Supplementing link text with the title attribute](https://www.w3.org/WAI/WCAG22/Techniques/html/H33)) is just one of several methods for identifying link purpose. The page that’s linked points out that there are more preferable ways of identifying link purpose:

> Because of the extensive user agent limitations in supporting access to the title attribute, authors should use caution in applying this technique. For this reason, it is preferred that the author use technique [C7: Using CSS to hide a portion of the link text](https://www.w3.org/WAI/WCAG22/Techniques/css/C7) or [H30: Providing link text that describes the purpose of a link for anchor elements](https://www.w3.org/WAI/WCAG22/Techniques/html/H30).

Technique H30 is currently possible with CKEditor, as authors can provide clear, descriptive link text. I could see something like being able to set an `aria-label` on an anchor tag; for example, a “Read more” link where an author could update the `aria-label` attribute to include clear information about the link purpose, like “Read more about growing potatoes.” However, as with any ARIA, there’s always the possibility that this could introduce more accessibility errors than it would fix.

It may be helpful to see what kind of rich text content you’re concerned may not meet the WCAG 2.2 AA conformance standard so we can better brainstorm a method for tackling that.

**Comment by christianluis** [comment]
@gcamacho079 Thank you for your suggestion. We face a similar situation in a project for the public sector. 

Your solutions unfortunately is not feasible for our project. Guidelines from this public client are based on WCAG but do explicitly demand for title and aria-labels within hyperlink-tags. 

Since you already provide a config method to add decorators, it would be handy to build up on that e.g. like:
```
{
  "link": {
    "decorators": {
      "addTitle": {
        "attributes": {
          "title": "$input"
        },
        "label": "Title",
        "mode": "input"
      }
    }
  },
  ...
}
```

**Comment by gcamacho079** [comment]
Thank you for adding to the discussion @christianluis. We’ve discussed this internally and plan to start working on this feature for an upcoming release.

**Comment by brandonkelly** [comment, marked as answer]
We released v5 of our CKEditor plugin today, with a new Link modal that reflects Craft’s Link fields, with the ability to also include advanced fields like “Title text”. (#375)
